src/components/moderation/moderation-common.ts

```diff
@@ -0,0 +1,163 @@
+import * as Discord from "discord.js";
+
+import { strict as assert } from "assert";
+
+import { critical_error, unwrap } from "../../utils.js";
+import { BotComponent } from "../../bot-component.js";
+import { TextBasedCommand } from "../../command.js";
+import { Wheatley } from "../../wheatley.js";
+
+import * as mongo from "mongodb";
+
+export type moderation_type = "mute" | "warn" | "ban" | "kick" | "no off-topic" | "rolepersist";
+
+export type moderation_entry = {
+    case_number: number;
+    user: string;
+    user_name: string;
+    moderator: string;
+    moderator_name: string;
+    type: moderation_type;
+    reason: string | null;
+    issued_at: number; // milliseconds since epoch
+    duration: number; // milliseconds
+    active: boolean;
+    removal_mod?: string;
+    removal_mod_name?: string;
+    removal_timestamp?: number; // milliseconds since epoch
+    removal_reason?: string | null;
+};
+
+export const duration_regex = /(?:perm\b|\d+\s*[mhdwMy])/;
+
+const INT_MAX = 0x7fffffff;
+
+export function parse_duration(duration: string) {
+    // TODO
```

Comment:
What format of duration is expected here? It would be nice if the TODO explained this, so that other contributors can provide the desired function.

Reply:
TODOs will be resolved here before anything is merged.